File: src/main/kotlin/org/kite9/diagram/common/elements/vertex/ContainerSideVertex.java
package org.kite9.diagram.common.elements.vertex;

import java.util.HashSet;
import java.util.Set;

import org.kite9.diagram.model.DiagramElement;
import org.kite9.diagram.visualization.planarization.mgt.ContainerConnectionTransform2;

/**
 * This is used by {@link ContainerConnectionTransform2}, which splits edges off from MultiCornerVertex points
 * onto their own vertices.
 * 
 * @author robmoffat
 */
public class ContainerSideVertex extends AbstractVertex implements MultiElementVertex {

	Set<DiagramElement> underlyings = new HashSet<>();	

	public ContainerSideVertex(String name) {
		super(name);
	}
	
	public ContainerSideVertex(String name, Set<DiagramElement> underlyings) {
		super(name);
		this.underlyings = underlyings;
	}

	@Override
	public boolean hasDimension() {
		return false;
	}

	@Override
	public boolean isPartOf(DiagramElement de) {
		return underlyings.contains(de);
	}
	
	public void addUnderlying(DiagramElement de) {
		if (de != null) {
			underlyings.add(de);
		}
	}

	@Override
	public Set<DiagramElement> getDiagramElements() {
		return underlyings;
	}
}

File: src/main/kotlin/org/kite9/diagram/common/elements/vertex/ContainerSideVertex.kt
package org.kite9.diagram.common.elements.vertex

import org.kite9.diagram.model.DiagramElement

/**
 * This is used by [ContainerConnectionTransform2], which splits edges off from MultiCornerVertex points
 * onto their own vertices.
 *
 * @author robmoffat
 */
class ContainerSideVertex(val id: String) : AbstractVertex(id), MultiElementVertex {

    private val underlyings : MutableSet<DiagramElement> = hashSetOf()

    override fun hasDimension(): Boolean {
        return false
    }

    override fun isPartOf(de: DiagramElement?): Boolean {
        return underlyings.contains(de)
    }

    fun addUnderlying(de: DiagramElement) {
        underlyings.add(de)
    }

    override fun getDiagramElements(): Set<DiagramElement> {
        return underlyings
    }
}